crates/paimon/src/spec/manifest_file_meta.rs

```diff
@@ -0,0 +1,184 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde_bytes::{Bytes, Deserialize as SerdeBytesDeserialize, Serialize as SerdeBytesSerialize};
+use std::fmt::{Display, Formatter};
+
+/// Metadata of a manifest file.
+///
+/// Impl Reference: <https://github.com/apache/paimon/blob/db8bcd7fdd9c2705435d2ab1d2341c52d1f67ee5/paimon-core/src/main/java/org/apache/paimon/manifest/ManifestFileMeta.java>
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize)]
+pub struct ManifestFileMeta {
+    /// manifest file name
+    #[serde(rename = "_FILE_NAME")]
+    file_name: String,
+
+    /// manifest file size.
+    #[serde(rename = "_FILE_SIZE")]
+    file_size: i64,
+
+    /// number added files in manifest.
+    #[serde(rename = "_NUM_ADDED_FILES")]
+    num_added_files: i64,
+
+    /// number deleted files in manifest.
+    #[serde(rename = "_NUM_DELETED_FILES")]
+    num_deleted_files: i64,
+
+    /// partition stats, the minimum and maximum values of partition fields in this manifest are beneficial for skipping certain manifest files during queries, it is a SimpleStats.
+    #[serde(rename = "_PARTITION_STATS")]
+    partition_stats: SimpleStats,
+
+    /// schema id when writing this manifest file.
+    #[serde(rename = "_SCHEMA_ID")]
+    schema_id: i64,
+}
+
+impl ManifestFileMeta {
+    /// Get the manifest file name
+    #[inline]
+    pub fn file_name(&self) -> &str {
+        self.file_name.as_str()
+    }
+
+    /// Get the manifest file size.
+    #[inline]
+    pub fn file_size(&self) -> i64 {
+        self.file_size
+    }
+
+    /// Get the number added files in manifest.
+    #[inline]
+    pub fn num_added_files(&self) -> i64 {
+        self.num_added_files
+    }
+
+    /// Get the number deleted files in manifest.
+    #[inline]
+    pub fn num_deleted_files(&self) -> i64 {
+        self.num_deleted_files
+    }
+
+    /// Get the partition stats
+    pub fn partition_stats(&self) -> &SimpleStats {
```

Comment:
`SimpleStats` could be `copy`, which can return `SimpleStats` directly.

Reply:
I think returning the reference is enough

Reply:
SimpleStats can't be `Copy` since it contains `Vec<u8>`.

Reply:
> SimpleStats can't be `Copy` since it contains `Vec<u8>`.

Sorry, my bad.